Gracefully handle error decoding cursor parameter

gherceg opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->

## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
Currently this results in a 500 which in addition to being gross to receive when using the API (we return the html sad danny page), should be gracefully handled on our end.

Here's the relevant [sentry error](https://dimagi.sentry.io/issues/6624687069/?project=136860&referrer=github-open-pr-bot). It isn't that common, but directly user facing and easy enough to fix.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
I double check that this is the error raised when failing to b64decode a value. We don't see it often, but it isn't helpful to the user currently.
### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
No

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

gherceg committed:
Gracefully handle error decoding cursor parameter

Currently this results in a 500 which in addition to being gross to
receive when using the API (we return the html sad danny page), should
be gracefully handled on our end.

esoergel reviewed (approved):
Huh, do you know why this happens?  Is it actually client error?

gherceg commented:
> Huh, do you know why this happens? Is it actually client error?

Yeah I initially thought it was an issue of us not url decoding the parameter first and created [this](https://github.com/dimagi/commcare-hq/pull/36500), but then realized that was not true. In my experience, this was because the output was truncated in my terminal for whatever reason and I ended up using an incomplete url/cursor parameter. I wasn't able to reproduce this error after realizing that so am chalking it up to user error.

gherceg committed:
Avoid attaching exception to reporting

The exception isn't easily serializable and also isn't necessary, so
let's just report the error and move on.
